Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -4,56 +4,59 @@ import Foundation
 /// The possible types of BSON values and their corresponding integer values.
 public enum BSONType: UInt32 {
     /// An invalid type
-    case invalid = 0x00,
+    case invalid = 0x00
     /// 64-bit binary floating point
-    double = 0x01,
+    case double = 0x01
     /// UTF-8 string
-    string = 0x02,
+    case string = 0x02
     /// BSON document
-    document = 0x03,
+    case document = 0x03
     /// Array
-    array = 0x04,
+    case array = 0x04
     /// Binary data
-    binary = 0x05,
+    case binary = 0x05
     /// Undefined value - deprecated
-    undefined = 0x06,
+    case undefined = 0x06
     /// A MongoDB ObjectId.
     /// - SeeAlso: https://docs.mongodb.com/manual/reference/method/ObjectId/
-    objectId = 0x07,
+    case objectId = 0x07
     /// A boolean
-    bool = 0x08,
+    case bool = 0x08
     /// UTC datetime, stored as UTC milliseconds since the Unix epoch
-    datetime = 0x09,
+    case datetime = 0x09
     /// Null value
-    null = 0x0a,
+    case null = 0x0a
     /// A regular expression
-    regex = 0x0b,
+    case regex = 0x0b
     /// A database pointer - deprecated
-    dbPointer = 0x0c,
+    case dbPointer = 0x0c
     /// Javascript code
-    code = 0x0d,
+    case code = 0x0d
     /// A symbol - deprecated
-    symbol = 0x0e,
+    case symbol = 0x0e
     /// JavaScript code w/ scope
-    codeWithScope = 0x0f,
+    case codeWithScope = 0x0f
     /// 32-bit integer
-    int32 = 0x10,
+    case int32 = 0x10
     /// Special internal type used by MongoDB replication and sharding
-    timestamp = 0x11,
+    case timestamp = 0x11
     /// 64-bit integer
-    int64 = 0x12,
+    case int64 = 0x12
     /// 128-bit decimal floating point
-    decimal128 = 0x13,
+    case decimal128 = 0x13
     /// Special type which compares lower than all other possible BSON element values
-    minKey = 0xff,
+    case minKey = 0xff
     /// Special type which compares higher than all other possible BSON element values
-    maxKey = 0x7f
+    case maxKey = 0x7f
 }
 
 /// A protocol all types representing `BSONType`s must implement.
-public protocol BSONValue {
+internal protocol BSONValue: Codable {
     /// The `BSONType` of this value.
-    var bsonType: BSONType { get }
+    static var bsonType: BSONType { get }
```

Comment:
each `BSONValue` corresponds to exactly one BSON type now, so this property can be static.